iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/metadata/ITableDeviceSchemaValidation.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.queryengine.plan.relational.metadata;
+
+import java.util.List;
+
+/**
+ * This class acts as a request for device schema validation and defines the necessary information
+ * interfaces.
+ */
+public interface ITableDeviceSchemaValidation {
+
+  String getDatabase();
+
+  String getTableName();
+
+  List<String[]> getDeviceIdList();
+
+  List<String> getAttributeColumnNameList();
+
+  List<List<String>> getAttributeValue();
```

Comment:
column-based or row-based?

Reply:
List<Object[]> will be better? If we extend attribute columns to allow other types.

Reply:
Fixed.

Reply:
Row based.